File: lambdify-apigateway-java/tests/lambdify/apigateway/URLTokenizerTest.java
package lambdify.apigateway;

import static org.junit.jupiter.api.Assertions.*;
import lombok.val;
import org.junit.jupiter.api.*;

public class URLTokenizerTest {

    @DisplayName("Can tokenize the root path")
    @Test void test1(){
        val tokens = URLMatcher.tokenize("/");
        assertFalse( tokens.isEmpty() );
        assertEquals( 1, tokens.size() );
        assertEquals( "/", tokens.get(0) );
    }

    @DisplayName( "Can tokenize the a multi-level path" )
    @Test void test2(){
        val tokens = URLMatcher.tokenize("/multi/level/path");
        assertFalse( tokens.isEmpty() );
        assertEquals( 3, tokens.size() );
        assertEquals( "/multi", tokens.get(0) );
        assertEquals( "/level", tokens.get(1) );
        assertEquals( "/path", tokens.get(2) );
    }

    @DisplayName( "Can tokenize the a multi-level path that ends with /" )
    @Test
    void test3(){
        val tokens = URLMatcher.tokenize("/multi/level/path");
        assertFalse( tokens.isEmpty() );
        assertEquals( 3, tokens.size() );
        assertEquals( "/multi", tokens.get(0) );
        assertEquals( "/level", tokens.get(1) );
        assertEquals( "/path", tokens.get(2) );
    }
    
}

File: lambdify-apigateway/tests/lambdify/apigateway/URLTokenizerTest.kt
package lambdify.apigateway

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

/**
 * Created by miere.teixeira on 09/04/2018.
 */
class URLTokenizerTest {

    val notFound = { _: APIGatewayRequest -> APIGatewayResponse.notFound() }
    val urlMatcher = UrlMatcher(notFound)

    @DisplayName("Can tokenize the root path")
    @Test fun test1(){
        val tokens = urlMatcher.tokenize("/")
        assertFalse( tokens.isEmpty() )
        assertEquals( 1, tokens.size )
        assertEquals( "/", tokens[0] )
    }

    @DisplayName( "Can tokenize the a multi-level path" )
    @Test fun test2(){
        val tokens = urlMatcher.tokenize("/multi/level/path")
        assertFalse( tokens.isEmpty() )
        assertEquals( 3, tokens.size )
        assertEquals( "/multi", tokens[0] )
        assertEquals( "/level", tokens[1] )
        assertEquals( "/path", tokens[2] )
    }

    @DisplayName( "Can tokenize the a multi-level path that ends with /" )
    @Test fun test3(){
        val tokens = urlMatcher.tokenize("/multi/level/path")
        assertFalse( tokens.isEmpty() )
        assertEquals( 3, tokens.size )
        assertEquals( "/multi", tokens[0] )
        assertEquals( "/level", tokens[1] )
        assertEquals( "/path", tokens[2] )
    }
}